includes/class-wc-order.php

```diff
@@ -2044,4 +2044,118 @@ public function get_order_item_totals( $tax_display = '' ) {
 
 		return apply_filters( 'woocommerce_get_order_item_totals', $total_rows, $this, $tax_display );
 	}
+
+	/**
+	 * Adds a '_held_for_checkout` record for all products in cart.
+	 *
+	 * @param WC_Cart $cart Cart instance.
+	 * @throws Exception When unable to hold stock for checkout.
+	 */
+	public function hold_stock_for_checkout( $cart ) {
+		if ( ! apply_filters( 'enable_hold_stock_3_9', true ) ) {
```

Comment:
It would introduce a bug, but this fix affects order creation flow which IMO is one of the most critical flow of the shop. So just in case, we missed something and it causes issues with someone's shop, I'd rather that they would have a way to disable it and live with the race condition bug instead of having problems in their order creation flow.